src/Modules/Admin/Commands.Admin/GetPowerBIActivityEvents.cs

```diff
@@ -0,0 +1,124 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Net;
+using Microsoft.PowerBI.Common.Api.ActivityEvent;
+using Microsoft.PowerBI.Common.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.PowerBI.Commands.Admin
+{
+    [Cmdlet(CmdletVerb, CmdletName)]
+    [OutputType(typeof(List<object>))]
+    public class GetPowerBIActivityEvents : PowerBIClientCmdlet
+    {
+        public const string CmdletName = "PowerBIActivityEvents";
+        public const string CmdletVerb = VerbsCommon.Get;
+        private bool validationError = false;
+
+        public GetPowerBIActivityEvents() : base() { }
+
+        public GetPowerBIActivityEvents(IPowerBIClientCmdletInitFactory init) : base(init) { }
+
+        [Parameter(Mandatory = true)]
+        public string StartDateTime { get; set; }
+
+        [Parameter(Mandatory = true)]
+        public string EndDateTime { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public string ActivityType { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public int ResultType { get; set; }
+
+        protected override void BeginProcessing()
+        {
+            base.BeginProcessing();
+            try
+            {
+                DateTime.Parse(this.StartDateTime);
+            }
+            catch
+            {
+                this.validationError = true;
+                this.Logger.ThrowTerminatingError($"{nameof(this.StartDateTime)} is not a valid DateTime.");
+            }
+
+            try
+            {
+                DateTime.Parse(this.EndDateTime);
+            }
+            catch
+            {
+                this.validationError = true;
+                this.Logger.ThrowTerminatingError($"{nameof(this.EndDateTime)} is not a valid DateTime.");
```

Comment:
We might want to pass an ErrorRecord instead to make it match closer to what PowerShell would throw when you passed in an invalid parameter